x/ecocredit/server/core/create_class.go

```diff
@@ -32,8 +32,20 @@ func (k Keeper) CreateClass(goCtx context.Context, req *core.MsgCreateClass) (*c
 		return nil, sdkerrors.ErrUnauthorized.Wrapf("%s is not allowed to create credit classes", adminAddress.String())
 	}
 
+	feeAmt := params.CreditClassFee.AmountOf(req.Fee.Denom)
+	if feeAmt.IsZero() {
+		return nil, sdkerrors.ErrInvalidRequest.Wrapf("%s is not allowed to be used in credit class fees", req.Fee.Denom)
+	}
+	reqFeeAmt, ok := sdk.NewIntFromString(req.Fee.Amount)
+	if !ok {
+		return nil, sdkerrors.ErrInvalidRequest.Wrapf("could not convert %s to %T", req.Fee.Amount, sdk.Int{})
+	}
+	if reqFeeAmt.LT(feeAmt) {
+		return nil, sdkerrors.ErrInsufficientFee.Wrapf("expected %v for fee, got %v", feeAmt, reqFeeAmt)
+	}
+
 	// Charge the admin a fee to create the credit class
-	err = k.chargeCreditClassFee(sdkCtx, adminAddress, params.CreditClassFee)
+	err = k.chargeCreditClassFee(sdkCtx, adminAddress, sdk.Coins{sdk.Coin{Denom: req.Fee.Denom, Amount: feeAmt}})
```

Comment:
force the fee to the amount specified by the params, rather than the amount passed, seemed like the proper thing to do?

Reply:
yes, related to my earlier review. I think we should adjust.